py4cast/datasets/poesy.py

```diff
@@ -58,118 +59,17 @@ def get_weight(level: float, level_type: str) -> float:
-    subgrid: Tuple[int] = (0, 0, 0, 0)
-    x: int = field(init=False)  # X dimension
-    y: int = field(init=False)  # Y dimension
-
-    def __post_init__(self):
-        ds = np.load(self.grid_filepath)
-        x, y = ds.shape
-        # Setting correct subgrid if no subgrid is selected.
-        if self.subgrid == (0, 0, 0, 0):
-            self.subgrid = (0, x, 0, y)
-        self.x = self.subgrid[1] - self.subgrid[0]
-        self.y = self.subgrid[3] - self.subgrid[2]
 
-    @property
-    def grid_filepath(self):
-        return SCRATCH_PATH / OROGRAPHY_FNAME
 
-    @cached_property
-    def lat(self) -> np.array:
-        filename = SCRATCH_PATH / LATLON_FNAME
-        with open(filename, "rb") as f:
-            lonlat = np.load(f)
-        return np.transpose(
-            lonlat[
-                1, self.subgrid[0] : self.subgrid[1], self.subgrid[2] : self.subgrid[3]
-            ]
-        )
-
-    @cached_property
-    def lon(self) -> np.array:
-        filename = SCRATCH_PATH / LATLON_FNAME
-        with open(filename, "rb") as f:
-            lonlat = np.load(f)
-        return lonlat[
-            0, self.subgrid[0] : self.subgrid[1], self.subgrid[2] : self.subgrid[3]
-        ]
-
-    @cached_property
-    def meshgrid(self) -> np.array:
-        """
-        Build a meshgrid from coordinates position.
-        """
-        return np.asarray([self.lat, self.lon])
-
-    @property
-    def geopotential(self) -> np.array:
-        ds = np.load(self.grid_filepath)
-        return ds[self.subgrid[0] : self.subgrid[1], self.subgrid[2] : self.subgrid[3]]
-
-    @property
-    def landsea_mask(self) -> np.array:
-        # TO DO : add landsea mask instead of orography
-        ds = np.load(self.grid_filepath)
-        return ds[self.subgrid[0] : self.subgrid[1], self.subgrid[2] : self.subgrid[3]]
-
-    @property
-    def border_mask(self) -> np.array:
-        if self.border_size > 0:
-            border_mask = np.ones((self.x, self.y)).astype(bool)
-            size = self.border_size
-            border_mask[size:-size, size:-size] *= False
-        elif self.border_size == 0:
-            border_mask = np.ones((self.x, self.y)).astype(bool) * False
-        else:
-            raise ValueError(f"Bordersize should be positive. Get {self.border_size}")
-        return border_mask
-
-    @property
-    def N_grid(self) -> int:
-        return self.x * self.y
-
-    @cached_property
-    def grid_limits(self):
-        return [  # In projection
-            self.lon[0, 0],  # min x
-            self.lon[-1, -1],  # max x
-            self.lat[0, 0],  # min y
-            self.lat[-1, -1],  # max y
-        ]
-
-    @cached_property
-    def projection(self):
-        # Create projection
-        return cartopy.crs.LambertConformal(central_longitude=2, central_latitude=46.7)
+def load_poesy_grid_info(grid: Grid) -> GridConfig:
+    geopotential = np.load(SCRATCH_PATH / OROGRAPHY_FNAME)
```

Comment:
Titan has its own var env `PY4CAST_TITAN_PATH`. Could Poesy have its own? And pass `OROGRAPHY_FNAME` and `LATLON_FNAME` as var env also ?

Reply:
ok, for now if you don't mind I just want to focus on the changes, then we will optimize POESY